### PR TITLE
specfile: Add support for EL8

### PR DESCRIPTION
**Pull request by faxm0dem**
List of changes:

* ganglia-devel is not available in el8 yet
* gpsd-devel is not available in el8 yet
* libmodbus-devel is not available in el8 yet
* liboping-devel is not available in el8 yet
* mosquitto-devel is not available in el8 yet
* fix python dep in el8
* disable liblvm
* dependency problem while configuring smart
* add check_uptime
* simplify python rhel version logic and force python3 for rhel8

Changelog: update specfile for EL8

Change-Id: I55323d7d8723a3030219a463314ec44c32d8df60

### CODE REVIEW

File: contrib/redhat/collectd.spec [hunk 1]
```diff
@@ -264,6 +266,18 @@
 %define with_xmms 0
 %endif
 
+# Plugins not buildable on RHEL 8
+%if 0%{?rhel} && 0%{?rhel} > 7
```
Comment (rubenk) [inline contrib/redhat/collectd.spec]: Purely cosmetic, but I'd prefer it if you used >= 8

File: contrib/redhat/collectd.spec [hunk 2]
```diff
@@ -823,11 +837,15 @@ Monitors process starts/stops via netlink library.
 Summary:	Python plugin for collectd
 Group:		System Environment/Daemons
 Requires:	%{name}%{?_isa} = %{version}-%{release}
+	%if 0%{?rhel} && 0%{?rhel} > 7
```
Comment (rubenk) [inline contrib/redhat/collectd.spec]: please left-align this

File: contrib/redhat/collectd.spec [hunk 3]
```diff
@@ -2045,6 +2075,7 @@ Collectd utilities
 	%{?_with_bind} \
 	%{?_with_ceph} \
 	%{?_with_cgroups} \
+	%{?_with_uptime} \
```
Comment (rubenk) [inline contrib/redhat/collectd.spec]: keep this sorted please

Comment (faxm0dem) [reply contrib/redhat/collectd.spec]: I don't :-)

```
☠ ./configure --help 2>&1 |grep uptime
  --enable-check_uptime   Notify about uptime reset
  --enable-uptime         Uptime statistics
```

**Comment by mkobyli** [comment]
Could you add ChangeLog line to your PR?

File: contrib/redhat/collectd.spec [hunk 4]
```diff
@@ -823,11 +837,15 @@ Monitors process starts/stops via netlink library.
 Summary:	Python plugin for collectd
 Group:		System Environment/Daemons
 Requires:	%{name}%{?_isa} = %{version}-%{release}
-	%if 0%{?rhel} && 0%{?rhel} < 6
+%if 0%{?rhel} && 0%{?rhel} > 7
```
Comment (mrunge) [inline contrib/redhat/collectd.spec]: this will also catch rhel8, but there is no python2.

Comment (faxm0dem) [reply contrib/redhat/collectd.spec]: are you saying python2 is only in centos?

Comment (mrunge) [reply contrib/redhat/collectd.spec]: python2 is in centos 7 (and RHEL7), python3 is in centos 8 and RHEL 8. There is no python2 in centos8 or RHEL8. The change will try to pull in python2 in cenots8 or rhel8.

Comment (faxm0dem) [reply contrib/redhat/collectd.spec]: Sorry I thought you meant there was a difference between centos8 and rhel8. Odd I tested this worked on centos8. I'm checking asap thanks

**Comment by faxm0dem** [comment]
so python2 is available in centos8 appstream.
what should we do?

**Comment by mrunge** [comment]
since python2 is dead (more or less), please try to move to python3.

**Comment by dago** [comment]
@faxm0dem Would you mind applying the discussed change? I would really like to include this PR in the upcoming 5.11 release.

**Comment by faxm0dem** [comment]
Sorry this takes so long. I'll do that tomorrow

**Comment by faxm0dem** [comment]
Sorry I got caught up into $LIFE I'll have another try on Thursday

**Comment by faxm0dem** [comment]
I rebased to 5.10.0 because 1c57925e6e60b326b4e96ea4f4c69dae58b34cd8 isn't in master. Any reason for this ?

**Comment by dago** [comment]
Not that I know of, would you mind cherrypicking this one into master?

**Comment by faxm0dem** [comment]
thing is, I don't have write access

**Comment by dago** [comment]
But you can create a PR from the commit to master which I can then merge 😄

**Comment by faxm0dem** [comment]
anything blocking this?

**Comment by mrunge** [comment]
Okay, two things are missing: a changelog line, @dago asked to apply this to master first, and travis CI is not reporting back.

**Comment by faxm0dem** [comment]
done

File: contrib/redhat/collectd.spec [hunk 5]
```diff
@@ -810,11 +824,22 @@ Monitors process starts/stops via netlink library.
...
+%if 0%{?rhel} == 8
+BuildRequires: python3-devel
+%endif
+%if 0%{?rhel} == 7
+BuildRequires: python2-devel
+%endif
+%if 0%{?rhel} == 6
+BuildRequires: python2-devel
+%endif
+%if 0%{?rhel} <= 5
```
Comment (rubenk) [inline contrib/redhat/collectd.spec]: RHEL 5 is eol so you can drop this. In that case you can reduce this down to python3-devel for rhel >= 8 and python2-devel for the rest

**Comment by mrunge** [comment]
there is still a changelog entry missing, and I'd prefer you'd also update this to 5.10, as the branch suggests it....

File: contrib/redhat/collectd.spec (hunk 5 above)
Comment (faxm0dem) [reply contrib/redhat/collectd.spec]: When rebasing to master my CI is failing on all RHEL versions, so don't merge yet. (this used to work on 5.10 branch)

**Comment by faxm0dem** [comment]
The changelog entry is there

File: contrib/redhat/collectd.spec [hunk 6]
```diff
@@ -262,6 +245,18 @@
...
+%define with_gmond 0
+%define with_gps 0
+%define with_modbus 0
+%define with_ping 0
+%define with_mqtt 0
+%endif
+
 Summary:	Statistics collection and monitoring daemon
 Name:		collectd
 Version:	5.9.2
```
Comment (mrunge) [inline contrib/redhat/collectd.spec]: 5.10.0

File: contrib/redhat/collectd.spec [hunk 7]
```diff
@@ -262,6 +245,18 @@
 %define with_xmms 0
 %endif
 
+# Plugins not buildable on RHEL 8
+%if 0%{?rhel} && 0%{?rhel} >= 8
```
Comment (mrunge) [inline contrib/redhat/collectd.spec]: %if 0%{?rhel } > 7

Comment (rubenk) [reply contrib/redhat/collectd.spec]: that's the same thing

Comment (rubenk) [reply contrib/redhat/collectd.spec]: but if you mean drop the first test and shorten this to `%if  0%{?rhel} >= 8` then yes, good idea :-)

**Comment by mrunge** [comment]
Could you please stop force-pushing changes? That makes it really hard to follow the changes.

You have dpdk_telemetry here, which is a 5.11 feature. At the same time, I requested to update at least top 5.10...

I still consider this as WIP, since there are still todos in the spec file, such as adding subpackages for buddyinfo etc.

File: contrib/redhat/collectd.spec (hunk 7 above)
Comment (faxm0dem) [reply contrib/redhat/collectd.spec]: I prefer to keep it that way. It was > before and someone asked me to use >= instead. ping pong

**Comment by faxm0dem** [comment]
sorry, but I'm requested to change things back and forth here (change target branch to 5.10 and back to master) etc. so I'm a bit lost.

I'm now working to make this ok on master, so that's where I'll merge it.
I can make another PR against 5.10 if you like

**Comment by faxm0dem** [comment]
also, there is no 5.10.0 tag so it's hard to know what actually is in 5.10.0

**Comment by faxm0dem** [comment]
so this now builds correctly on el6,7 and 8

**Comment by rubenk** [comment]
@mrunge this seems good to merge?

**Comment by faxm0dem** [comment]
thanks, and sorry for the noise
